File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = 1
        seen = {}

        right = 0
        left = 0

        while right < len(s):
            seen[s[right]] = 1 + seen.get(s[right], 0)
            replaced = right - left - max(seen.values()) + 1

            while replaced > k and left < right:
                seen[s[left]] = seen.get(s[left], 1) - 1
                left += 1
                
                replaced = right - left - max(seen.values()) + 1

            right += 1
            res = max(res, right - left)

        return res

File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;

        while (left <= right) {
            int idx = (right + left) / 2;

            if (nums[idx] < target)
                left = idx + 1;
            else if (nums[idx] > target)
                right = idx - 1;
            else
                return idx;
        }

        return -1;
    }
};
